src/pymorize/global_attributes.py

```diff
@@ -0,0 +1,252 @@
+# global_attributes.py
+
+import re
+from datetime import datetime
+
+_parent_fields = (
+    "branch_method",
+    "branch_time_in_child",
+    "branch_time_in_parent",
+    "parent_experiment_id",
+    "parent_mip_era",
+    "parent_source_id",
+    "parent_time_units",
+    "parent_variant_label",
+)
+
+
+"""
+attribute dependencies
+----------------------
+Table header
+------------
+data_specs_version
+Conventions
+mip_era
+realm
+product
+frequency
+
+CV
+---
+source_id  <user input>
+    source
+    institution_id
+    license_info
+    model_component  # how to get model_component <user input>
+        native_nominal_resolution (nominal_resolution)
+        description (grid)
+experiment_id
+    activity_id
+    parent_experiment_id
+    sub_experiment_id
+
+User input
+----------
+table_id
+further_info_url
+institution
+variant_label
+    initialization_index
+    realization_index
+    forcing_index
+    physics_index
+
+system generated
+----------------
+creation_date
+tracking_id
+"""
+
+
+class GlobalAttributes:
+    def __init__(self, cv):
+        """
+        Parameters
+        ----------
+        cv : ControlledVocabularies
+            Controlled Vocabularies for CMIP6.
+        """
+        self.cv = cv
+
+    def _parse_variant_label(self, label: str) -> dict:
+        """
+        Extracts indices values from variant label.
+        `label` must be of the form "r<int>i<int>p<int>f<int>".
+        Example
+        -------
+        >>> label = "r1i1p1f1"
+        >>> _parse_variant_label(label)
+        {'realization_index': 1, 'initialization_index': 1, 'physics_index': 1, 'forcing_index': 1,}
+        """
+        pattern = re.compile(
+            r"r(?P<realization_index>\d+)"
+            r"i(?P<initialization_index>\d+)"
+            r"p(?P<physics_index>\d+)"
+            r"f(?P<forcing_index>\d+)"
+            r"$"
+        )
+        if label is None:
+            raise ValueError(
+                f"`label` must be of the form 'r<int>i<int>p<int>f<int>', Got: {label}"
+            )
+        d = pattern.match(label)
+        if d is None:
+            raise ValueError(
+                f"`label` must be of the form 'r<int>i<int>p<int>f<int>', Got: {label}"
+            )
+        d = {name: int(val) for name, val in d.groupdict().items()}
+        return d
+
+    def _source_id_related(self, rule: dict) -> dict:
```

Comment:
Please add a doc-string and some examples here, this is a big enough function to deserve some testing.